### PR TITLE
improve extract_number_en

### PR DESCRIPTION
**Pull request by JarbasAl**
## Description
when using extractnumber i noticed "twenty two" would not work... it is great with fractions, but fails on big numbers, i assume because it was meant to handle time

this PR adds support for negative numbers, numbers with decimal points and big numbers in general

https://en.wikipedia.org/wiki/Names_of_large_numbers

## How to test

check the unittests

## Contributor license agreement signed?
CLA [ yes] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/util/lang/format_en.py [hunk 1]
```diff
@@ -17,6 +17,105 @@
 
 from mycroft.util.lang.format_common import convert_to_mixed_fraction
 
+LONG_SCALE_EN = {
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply mycroft/util/lang/format_en.py]: go home codacy, you are drunk

File: mycroft/util/lang/format_en.py [hunk 2]
```diff
@@ -17,6 +17,105 @@
...
+    10e903: "trecentillion",
+    10e1203: "quadringentillion",
+    10e1503: "quingentillion",
+    10e1803: "sescentillion",
+    10e2103: "septingentillion",
+    10e2403: "octingentillion",
+    10e2703: "nongentillion",
+    10e3003: "millinillion",
+    10e100: "googol"
+}
```
[the remark this answers is not in the document]
Comment (JarbasAl) [reply mycroft/util/lang/format_en.py]: i put them there because i think these will be used in pronounce_number functions (that's for another PR)

**Comment by JarbasAl** [comment]
short scale and long scale are different naming schemes, 

a Billion can mean 10e9 or 10e12, more info the wiki link

**Comment by JarbasAl** [comment]
i am working on a related [PR for pronounce number](https://github.com/JarbasAl/jarbas-core/blob/feature_pronounce_number/mycroft/util/lang/format_en.py), missing long scale for now

some tests:
```
    def test_convert_hundreds(self):
        self.assertEqual(pronounce_number(100), "one hundred")
        self.assertEqual(pronounce_number(666), "six hundred and sixty six")
        self.assertEqual(pronounce_number(1456), "one thousand, four hundred "
                                                  "and fifty six")
        self.assertEqual(pronounce_number(103254654), "one hundred and three "
                                                      "million, two hundred "
                                                      "and fifty four "
                                                      "thousand, six hundred "
                                                      "and fifty four")
        self.assertEqual(pronounce_number(1512457), "one million, five hundred"
                                                    " and twelve thousand, "
                                                    "four hundred and fifty "
                                                    "seven")
        self.assertEqual(pronounce_number(209996), "two hundred and nine "
                                                   "thousand, nine hundred "
                                                   "and ninety six")
        self.assertEqual(pronounce_number(95505896639631893),
                         "ninety five quadrillion, five hundred and five "
                         "trillion, eight hundred and ninety six billion, six "
                         "hundred and thirty nine million, six hundred and "
                         "thirty one thousand, eight hundred and ninety three")

```

**Comment by forslund** [comment]
Merging after tiny addition.

**Pull request merged**